src/app/pages/storage/volumes/datasets/dataset-form/dataset-form.component.ts

```diff
@@ -1466,34 +1468,22 @@ export class DatasetFormComponent implements FormConfiguration {
       delete data.special_small_block_size;
     }
 
-    if (data.quota_warning_inherit) {
-      delete (data.quota_warning);
+    if (data.quota_warning_inherit && this.dataset?.quota_warning) {
```

Comment:
Are the same changes required in `addSubmit`?

Reply:
No need to change in `addSubmit`
For example, if `quota_warning_inherit` is true, just need only to delete `quota_warning` in `addSubmit`. but need to submit value `INHERIT` if exists origin value of `quota_warning` in `editSubmit`